staging/src/k8s.io/kms/internal/mock_latency_remote_service_test.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+import (
+	"bytes"
+	"context"
+	"math/rand"
+	"testing"
+	"time"
+
+	"k8s.io/kms/pkg/service"
+)
+
+const (
+	testLatencyInMillisecond = 100
```

Comment:
This is basically guaranteed to flake in CI.  You probably want to set the sleep to `3s` and then just check if the `time.Since` is greater than `1s` and less then `5s`

Reply:
I think setting a `3s` sleep directly conflicts with the 1 qps we are setting for the rate limiter.  anything longer than `1s` sleep ensures the qps is no more than 1. I _think_ this is how NewTokenBucketRateLimiter works.